Move initial WebRTC demo

jbolinger opened the pull request.

idzkowski-google commented on WebRTCSample/app/res/xml/preferences.xml:
```diff
@@ -0,0 +1,252 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/values/styles.xml:
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/values/strings.xml:
```diff
@@ -0,0 +1,230 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/values/arrays.xml:
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/mipmap-anydpi-v26/ic_launcher_round.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/mipmap-anydpi-v26/ic_launcher.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/menu/connect_menu.xml:
```diff
@@ -0,0 +1,14 @@
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/layout/glass_fragment_hud.xml:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/layout/glass_fragment_call.xml:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/layout/glass_activity_connect.xml:
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/res/drawable/ic_launcher_foreground.xml:
```diff
@@ -0,0 +1,12 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/build.gradle:
```diff
@@ -0,0 +1,56 @@
+apply plugin: 'com.android.application'
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/AndroidManifest.xml:
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="utf-8"?>
```
nit: Missing header

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/FileDataSaver.java:
```diff
@@ -0,0 +1,109 @@
+    public FileDataSaver(FileCreator fileCreator) {
+        file = fileCreator.createFile();
+        separator = System.getProperty("line.separator");
+
```
nit: Empty line

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/BaseCallActivity.java:
```diff
@@ -0,0 +1,868 @@
+import org.webrtc.VideoFrame;
+import org.webrtc.VideoSink;
+
+public abstract class BaseCallActivity extends Activity implements AppRTCClient.SignalingEvents,
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/BaseConnectActivity.java:
```diff
@@ -0,0 +1,417 @@
+import java.util.Random;
+
+
+public abstract class BaseConnectActivity extends Activity {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/FileSaverFactory.java:
```diff
@@ -0,0 +1,10 @@
+package org.appspot.apprtcstandalone;
+
+public class FileSaverFactory {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java:
```diff
@@ -0,0 +1,62 @@
+import com.example.glass.ui.GlassGestureDetector.OnGestureListener;
+import org.webrtc.StatsReport;
+
+public class GlassCallActivity extends BaseCallActivity implements OnGestureListener {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:
```diff
@@ -0,0 +1,64 @@
+import javax.annotation.Nullable;
+import org.webrtc.RendererCommon.ScalingType;
+
+public class GlassCallFragment extends Fragment {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java:
```diff
@@ -0,0 +1,19 @@
+package org.appspot.apprtcstandalone;
+
+public class Resolution {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java:
```diff
@@ -0,0 +1,22 @@
+package org.appspot.apprtcstandalone;
+
+public class SimpleCaptureOptions implements CaptureOptionsPicker.CaptureOptions {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java:
```diff
@@ -0,0 +1,19 @@
+
+import java.util.Random;
+
+public class RandomRoomNameCreator implements RoomNameCreator {
```
nit: Missing javadoc for public class

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RoomNameCreator.java:
```diff
@@ -0,0 +1,5 @@
+package org.appspot.apprtcstandalone.util;
+
+public interface RoomNameCreator {
```
nit: Missing javadoc for public interface

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java:
```diff
@@ -0,0 +1,19 @@
+
+    @Override
+    public String getRoomName() {
+        Random random = new Random(System.currentTimeMillis());
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/util/RandomRoomNameCreator.java:
```diff
@@ -0,0 +1,19 @@
+    @Override
+    public String getRoomName() {
+        Random random = new Random(System.currentTimeMillis());
+        StringBuilder roomName = new StringBuilder();
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java:
```diff
@@ -0,0 +1,22 @@
+
+public class SimpleCaptureOptions implements CaptureOptionsPicker.CaptureOptions {
+
+    private Resolution resolution;
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SimpleCaptureOptions.java:
```diff
@@ -0,0 +1,22 @@
+public class SimpleCaptureOptions implements CaptureOptionsPicker.CaptureOptions {
+
+    private Resolution resolution;
+    private int framerate;
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:
```diff
@@ -0,0 +1,79 @@
+     * @param fileName      the name of the file to be appended with date of file creation.
+     */
+    public SdCardFileCreator(CharSequence directoryName, CharSequence fileName) {
+
```
nit: Empty line

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:
```diff
@@ -0,0 +1,79 @@
+
+    @Override
+    public File createFile() {
+        File directory =
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:
```diff
@@ -0,0 +1,79 @@
+            }
+        }
+
+        File file =
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    private String getCreationDate() {
+        StringBuilder outputDate = new StringBuilder();
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:
```diff
@@ -0,0 +1,79 @@
+
+    private String getCreationDate() {
+        StringBuilder outputDate = new StringBuilder();
+        Calendar cal = Calendar.getInstance();
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/SdCardFileCreator.java:
```diff
@@ -0,0 +1,79 @@
+
+    private String getFilePath(CharSequence directory) {
+        return SD_CARD_DIR + SLASH + directory;
+
```
nit: Empty line

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java:
```diff
@@ -0,0 +1,19 @@
+package org.appspot.apprtcstandalone;
+
+public class Resolution {
+    private int height;
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/Resolution.java:
```diff
@@ -0,0 +1,19 @@
+
+public class Resolution {
+    private int height;
+    private int width;
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java:
```diff
@@ -0,0 +1,29 @@
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_hud, container, false);
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java:
```diff
@@ -0,0 +1,29 @@
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_hud, container, false);
+    TextView roomName = controlView.findViewById(R.id.room_name_textview);
+    roomName.setText(getArguments().getString(GlassCallActivity.EXTRA_ROOMID));
```
nit: Should be moved to the onViewCreated method

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassHudFragment.java:
```diff
@@ -0,0 +1,29 @@
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_hud, container, false);
+    TextView roomName = controlView.findViewById(R.id.room_name_textview);
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java:
```diff
@@ -0,0 +1,140 @@
+
+  private void resetViews() {
+    View mainView = getLayoutInflater().inflate(R.layout.glass_activity_connect, null, false);
+    setContentView(mainView);
```
nit: Should be moved to onCreate

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java:
```diff
@@ -0,0 +1,140 @@
+
+  private void refreshUi() {
+    isConnecting = true;
+    ViewGroup mainView = findViewById(android.R.id.content);
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:
```diff
@@ -0,0 +1,64 @@
+  @Override
+  public void onStart() {
+    super.onStart();
+    DisplayMetrics displayMetrics = new DisplayMetrics();
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:
```diff
@@ -0,0 +1,64 @@
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_call, container, false);
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallFragment.java:
```diff
@@ -0,0 +1,64 @@
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View controlView = inflater.inflate(R.layout.glass_fragment_call, container, false);
+    captureFormatText = controlView.findViewById(R.id.capture_format_text_call);
+    captureFormatSlider = controlView.findViewById(R.id.capture_format_slider_call);
```
nit: Should be moved to the onViewCreated method

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java:
```diff
@@ -0,0 +1,62 @@
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    DisplayMetrics displayMetrics = new DisplayMetrics();
```
nit: Can be final

idzkowski-google commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassCallActivity.java:
```diff
@@ -0,0 +1,62 @@
+    gestureDetector = new GlassGestureDetector(this, this);
+
+    callFragment.setArguments(getIntent().getExtras());
+    FragmentTransaction ft = getFragmentManager().beginTransaction();
```
nit: Can be final

jbolinger commented on WebRTCSample/app/src/org/appspot/apprtcstandalone/GlassConnectActivity.java:
```diff
@@ -0,0 +1,140 @@
+package org.appspot.apprtcstandalone;
```
Added (and same for other `Glass*`) files. We shouldn't add headers to the existing code that was modified from the example.

jbolinger commented:
Thanks @idzkowski-google - we don't want to mess with all of the code that was forked so I'll leave those unresolved. I fixed a few that were `Glass*` specific and added a copy of the original license to the readme so it's clearer.
